### ISSUE TITLE
AnalysisRun final error message for metrics that failed to run, is very long and repeating itself

### ISSUE THREAD

**Issue by caduri** [argoproj/argo-rollouts#2258]
**Describe the bug**

When a Prometheus metric is failing due to a bad address then the error message in AnalysisRun is long and repeating itself
```
Metric assessed Error due to consecutiveErrors (5) >
consecutiveErrorLimit (4): "Error Message: Metric assessed Error due to
consecutiveErrors (5) > consecutiveErrorLimit (4): "Error Message:
Metric assessed Error due to consecutiveErrors (5) >
consecutiveErrorLimit (4): "Error Message: Metric assessed Error due to
```
        
note: This is just an example the error message is longer

**To Reproduce**

In order to reproduce:
1. Create ClusterAnalysisTemplate with bad prometheus address
2. Add it to a rollout strategy as step analysis
3. Check the error message in AnalysisRun onces its finish running

**Expected behavior**

A readable error message

**Screenshots**

<img width="1095" alt="Screen Shot 2022-09-22 at 15 59 23" src="https://user-images.githubusercontent.com/4005134/191753587-1be8ec9e-9f6e-4b81-9a1a-f83923c08c4f.png">

note: if you look at the mini map you can see that the message is very long

**Version**

I tried it on both versions v1.3.0 and v1.2.1

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by iam-veeramalla** [comment]
@caduri I was trying to reproduce the issue at my end. 

When I submit a bad Prometheus address, I ran into a different error which looks good.
```
      message: 'Post "http://prometheus.example.com:9090/api/v1/query": dial tcp:
        lookup prometheus.example.com on 10.96.0.10:53: no such host'
      phase: Error
```

I might be missing something, Does the `bad Prometheus address` mean an `invalid address` ? To reproduce the above scenario I used `prometheus.example.com` which does not exist.

**Comment by zachaller** [comment]
I was able to reproduce this on 1.3.1 it is fixed in 1.3.2 and 1.4.0, can you try the new versions and reopen if you still see this
